pkg/kubelet/kubelet_pods.go

```diff
@@ -891,12 +892,17 @@ func (kl *Kubelet) getPullSecretsForPod(pod *v1.Pod) []v1.Secret {
 		secret, err := kl.secretManager.GetSecret(pod.Namespace, secretRef.Name)
 		if err != nil {
 			klog.InfoS("Unable to retrieve pull secret, the image pull may not succeed.", "pod", klog.KObj(pod), "secret", klog.KObj(secret), "err", err)
+			failedSecrets = append(failedSecrets, secretRef.Name)
 			continue
 		}
 
 		pullSecrets = append(pullSecrets, *secret)
 	}
 
+	if len(failedSecrets) != 0 {
+		kl.recorder.Eventf(pod, v1.EventTypeWarning, "FailedToRetrieveImagePullSecret", "Unable to retrieve image pull secret %s, the image pull may not succeed.", failedSecrets)
+	}
```

Comment:
as `failedSecrets` is a slice type, the `Eventf` output may looks like `image pull secret [secretA, secretB], the image...`. We can refactor with `strings.Join(failedSecrets, ", ")`. I think it's prettier.

Reply:
Yes, I was careless.

Reply:
Another advice, rebase multi commits to one commit as they are doing the same thing, we can keep the baseline of master branch clearly.